### ISSUE TITLE
About torch.backends.cudnn.deterministic issue

### ISSUE THREAD

**Issue by konioyxgq** [pytorch/pytorch#40134]
In order to reproduce the training process, I set torch.backends.cudnn.deterministic to FALSE, but this slowed down for almost an hour. Is there any way to reproduce the training process under the condition of torch.backends.cudnn.deterministic for TRUE. Do you have any suggestions？

**Comment by ptrblck** [comment]
Hi @konioyxgq,

could you explain your use case a bit?
Based on your description, you had to set `cudnn.deterministic = False` to get reproducible results, which also results in a slowdown or do you have a typo in the setup?
Also, what does the slowdown of an hour refer to? Is it for one epoch or the complete training, and what was the timing before?

`cudnn.deterministic = True` should give reproducible results (with the exceptions mentioned in the [Reproducibility docs](https://pytorch.org/docs/stable/notes/randomness.html)) and might result in a slowdown as a trade-off. We recommend to use the latest cudnn version

**Comment by konioyxgq** [comment]
Sorry, I wrote it wrong, it's **torch.backends.cudnn.benchmark** instead of **torch.backends.cudnn.deterministic**.
In order to make the training process reproducible, I made the following settings：
```
    torch.manual_seed(SEED)
    torch.cuda.manual_seed_all(SEED)
    torch.backends.cudnn.benchmark = False
    torch.backends.cudnn.deterministic = True
```
Training according to this setting, an epoch takes about 4 and a half hours.
But I only changed **torch.backends.cudnn.benchmark** to True. The whole training process only takes 3 and a half hours.
Can I ensure that the training is reproducible when **torch.backends.cudnn.benchmark** is set to ture? If not, what should I do to further speed up the training? do you have any good advice?

**Comment by danieltudosiu** [comment]
It also happens to me on one of my networks. I am using torch=1.7.0+cu110 and I set my seed using the monai.utils.set_determinism(seed=config["seed"]) which does [the following](https://docs.monai.io/en/latest/_modules/monai/utils/misc.html#set_determinism).

**Comment by veb-101** [comment]
> It also happens to me on one of my networks. I am using torch=1.7.0+cu110 and I set my seed using the monai.utils.set_determinism(seed=config["seed"]) which does [the following](https://docs.monai.io/en/latest/_modules/monai/utils/misc.html#set_determinism).
> 
> Weirdly enough, with another network, I do not get the huge slowdown. Doing a quick and dirty benchmark I get the following estimated epoch times:
> 
> * Problematic network:
>   
>   * Benchmark=True, Deterministic=False ~ 10m
>   * Benchmark=False, Deterministic=False ~ 18m
>   * Benchmark=False, Deterministic=True ~ 4h 23m
> * Normal network:
>   
>   * Benchmark=True, Deterministic=False ~ 8m
>   * Benchmark=False, Deterministic=False ~ 10m
>   * Benchmark=False, Deterministic=True ~ 12m
> 
> I have tried the quick and dirty benchmark on torch==1.7.0+cu110, 1.7.1, 1.7.1+cu110, and got the same behaviors.

I'm experiencing the same issue, yesterday during training, each epoch took around 15mins max. Today I looked at some resources for reproducibility and set the bechmark=False and deterministic=True and it's taking more than an hour for each epoch.

**Comment by konioyxgq** [comment]
When bechmark=True, cudnn searches for the fastest algorithm to do CNN, so the time difference is much

**Comment by SiminLi** [comment]
I have the same issue. I was running a wavenet-based model (mainly stacked 1D dilated convolution). With torch.backends.cudnn.deterministic=True and torch.backend.cudnn.benchmark=False, one epoch is ~379 second, without that two lines one epoch is 36 second/epoch. Believe it's a bug and seeking solutions here.

**Comment by YiheWang** [comment]
I have the same issue. I was setting torch.backends.cudnn.deterministic=True, and it extremely slowed down the training process. For some networks structure, I simply comment out this line of code, and the result is still reproducible, but not always.
